rfc/transact_from_substrate_to_ethereum.md

```diff
@@ -0,0 +1,55 @@
+# RFC: Transact from Substrate to Ethereum
+
+
+## Summary
+
+This RFC proposes the feature to call transact from Substrate to Ethereum through our bridge, including two PRs separately with https://github.com/Snowfork/snowbridge/pull/1145 for solidity and https://github.com/Snowfork/polkadot-sdk/pull/116 for substrate.
+
+## Explanation
+
+We use penpal for the integration, basically it works as follows:
```

Comment:
Fee flow:

- User represents a user who kicks off an extrinsic on the parachain.
- Parachain represents the source parachain, its sovereign or its agent depending on context.

Sequence|Where|Who|What
-|-|-|-
1|Parachain|User|pays(DOT, Native) to node to execute custom extrinsic; pays (DOT) to Treasury for delivery as part of custom extrinsic.
2|Bridge Hub|Parachain|pays(DOT) to Treasury Account for delivery(local fee), pays(DOT) to Parachain sovereign for delivery(remote fee), essentially a refund. Remote fee converted to ETH here.
3|Gateway|Relayer|pays(ETH) to validate and execute message.
4|Gateway|Parachain Agent|pays(ETH) to relayer for delivery(reward+refund) and execution.

Does this balance? No, Parachain agent pays for delivery and execution on the Ethereum side. User only pays for calling the extrinsic and delivery on the source Parachain. Because the custom extrinsic essentially allows the user to impersonate the parachain on ethereum side it becomes the Parachains job to charge appropriately for both delivery and execution. This doesn't mean there is something wrong with the implementation, just that this needs to be communication to creator of the custom extrinsic.

Reply:
For step 2 only pays(DOT) for the delivery(local fee) portion.

https://github.com/Snowfork/polkadot-sdk/blob/569d638d429944fc8d4e2d7db3b653d565879f86/bridges/snowbridge/primitives/router/src/outbound/mod.rs#L119-L125

Reply:
Summarize it in https://github.com/Snowfork/snowbridge/pull/1145/commits/bd88ed892f307ac79a941f0d45097ba17a8b47ee